Plugins/Player/Player.cpp

```diff
@@ -1336,4 +1344,167 @@ ArgumentStack Player::SetResManOverride(ArgumentStack&& args)
+        
+        if (pCreature && pCreature->m_pJournal)
+        {
+            auto entries = pCreature->m_pJournal->m_lstEntries;
+            
+            const auto questName = Services::Events::ExtractArgument<std::string>(args);
+            const auto questText = Services::Events::ExtractArgument<std::string>(args);
+            const auto tag = Services::Events::ExtractArgument<std::string>(args);
+            
+            ASSERT_OR_THROW(tag.size() > 0);
+            
+            const auto state = Services::Events::ExtractArgument<int32_t>(args);
+            const auto priority = Services::Events::ExtractArgument<int32_t>(args);
+            const auto completed = Services::Events::ExtractArgument<int32_t>(args);
+            const auto diplayed = Services::Events::ExtractArgument<int32_t>(args);
+            const auto updated = Services::Events::ExtractArgument<int32_t>(args);
+            
+            auto calDay = Services::Events::ExtractArgument<int32_t>(args);
+            auto timeDay = Services::Events::ExtractArgument<int32_t>(args);
+            auto silentUpdate = Services::Events::ExtractArgument<int32_t>(args);
+            
+            ASSERT_OR_THROW(state >= 0);
+            ASSERT_OR_THROW(priority >= 0);
+            ASSERT_OR_THROW(completed >= 0);
+            ASSERT_OR_THROW(updated >= 0);
+            ASSERT_OR_THROW(silentUpdate >= 0);
+            
+            
+            //If server owner leaves this 0 - the entry will be added with todays date
+            if (calDay <= 0)
+            {
+                calDay = Globals::AppManager()->m_pServerExoApp->GetWorldTimer()->GetWorldTimeCalendarDay();
+            }
+            //If server owner leaves this 0 - the entry will be added with now() time
+            if(timeDay <= 0)
+            {
+                timeDay = Globals::AppManager()->m_pServerExoApp->GetWorldTimer()->GetWorldTimeTimeOfDay();
+            }
+            
+            SJournalEntry newJournal;
+            newJournal.szName       = Utils::CreateLocString(questName,0,0);
+            newJournal.szText       = Utils::CreateLocString(questText,0,0);
+            newJournal.nCalendarDay = calDay;
+            newJournal.nTimeOfDay   = timeDay;
+            newJournal.szPlot_Id    = CExoString(tag.c_str());
+            newJournal.nState       = state;
+            newJournal.nPriority    = priority;
+            newJournal.nPictureIndex= 0; // Not implemented by bioware/beamdog  
+            newJournal.bQuestCompleted= completed; 
+            newJournal.bQuestDisplayed= diplayed; 
+            newJournal.bUpdated     = updated; 
+                   
+            auto *pMessage = static_cast<CNWSMessage*>(Globals::AppManager()->m_pServerExoApp->GetNWSMessage());       
+            if (pMessage)
+                {
+                 
+                    int overwrite = -1;
```

Comment:
It looks like you are using overwrite as a bool instead of an index, the `overwrite = i;` isn't used. I'd rewrite the loop as:

``` cpp
int i = 0;
for (i = 0; i < entries.num; i++)
{
    auto* pEntry = entries.element[i];
    if (pEntry->szPlot_Id.CStr() == tag)
    {
        pCreature->m_pJournal->m_lstEntries[i] = newJournal;
        break;
    }
}
if(i == entries.num)
    pCreature->m_pJournal->m_lstEntries.Add(newJournal);

```

Reply:
Had some issues implementing the approach above 'exactly'
Merged the main principle described above into the existing for loop structure.